src/ir/ctx.rs

```diff
@@ -1,21 +1,34 @@
 use super::Component;
-use crate::utils::Idx;
+use crate::utils::{Idx, IdxLike};
 
 /// A context for storing values with their indices.
-/// The context is indexed by [`Idx<T>`].
-pub trait Ctx<T> {
+/// The context is indexed by [`IdxLike<T>`].
+pub trait AddCtx<T, I = Idx<T>>
```

Comment:
Oh cool, I like separating this out. Was thinking of doing this with the mono wrappers so I'll use this once it's merged